src/marshmallow/fields.py

```diff
@@ -566,6 +571,8 @@ def schema(self):
             context = getattr(self.parent, "context", {})
             if callable(self.nested) and not isinstance(self.nested, type):
                 nested = self.nested()
+            elif isinstance(self.nested, dict):
+                nested = self._nested_from_dict(self.nested)
```

Comment:
What about a callable returning a dict?

Reply:
It would be very easy to support. (Just push this down a couple of lines and update the type annotation.)

If I do that, the `:param nested:` doc needs an update. Would the following work?

> `Schema` instance, class, class name (string), dictionary, or callable that returns a `Schema` or dictionary. Dictionaries are converted with `Schema.from_dict`.

Reply:
Yes. Not that the use case is that important but when I looked at the code I figured it wouldn't be more complicated and I saw no reason not to do it.